File: 백준/Bronze/10809. 알파벳 찾기/알파벳 찾기.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('');

const ABC = ['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z'];

solution(input);

function solution(S) {
    let answer = [-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1];
    
    for(let i =0; i<S.length; i++){
        for(let j = 0; j<ABC.length; j++){
            if(S[i] === ABC[j]){
                if(answer[j] === -1){
                answer[j] = i;                    
                }
            }
        }
    }
    console.log(...answer);
}

File: 백준/Bronze/10809. 알파벳 찾기/알파벳 찾기.java
import java.io.*;
import java.util.*;

public class Main {

	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		StringBuilder sb = new StringBuilder();

		String[] S = br.readLine().split("");

		String[] alph = { "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s",
				"t", "u", "v", "w", "x", "y", "z" };

		int[] countArr = new int[26];

		for (int j = 0; j < 26; j++) {
			if (Arrays.asList(S).contains(alph[j])) {
				countArr[j] = Arrays.asList(S).indexOf(alph[j]);
			} else {
				countArr[j] = -1;
			}
		}

		for (int i = 0; i < 26; i++)
			sb.append(countArr[i]).append(" ");

		bw.write(sb.toString().trim());
		bw.flush();
		bw.close();
		br.close();
	}

}